### PR TITLE
Ruby group capstone - Catalog of my things

### PR DESCRIPTION
**Pull request by nahnahdev**
## Project requirements

1. **Logic**

- Each class from the [UML class diagram](https://github.com/microverseinc/curriculum-ruby/blob/main/group-capstone/images/catalog_of_my_things.png) should be created in a separate .rb file.

- All associations (1-to-many relationships) and aggregations (parent-child relationships) between classes have been implemented.

- All properties visible in the diagram were defined and set up in the constructor method. Exception: properties for the 1-to-many relationships. These were NOT set in the constructor method. Instead, they were provided a custom setter method.

- All methods visible in the diagram have been implemented:

**add_item method (in all classes that have that method)**

> should take an instance of the Item class as an input.
> should add the input item to the collection of items.
> should add self as a property of the item object (by using the correct setter from the item object).

**can_be_archived?() in the Item class**

> should return true if published_date is older than 10 years.
> otherwise, it should return false.

**move_to_archive() in the Item class**

> should reuse can_be_archived?() method.
> should change the archived property to true if the result of the can_be_archived?() method is true.
> should do nothing if the result of the can_be_archived?() method is false.

**can_be_archived?() in the Book class**

> should override the method from the parent class.
> should return true if the parent's method returns true OR if cover_state equals "bad".
> otherwise, it should return false.

**can_be_archived?() in the MusicAlbum class**

> should override the method from the parent class.
> should return true if the parent's method returns true AND if on_spotify equals true.
> otherwise, it should return false.

**can_be_archived?() in the Game class**

> should override the method from the parent class.
> should return true if the parent's method returns true AND if last_played_at is older than 2 years.
> otherwise, it should return false.

2. **Add unit tests for all implemented methods.**

**User interface**

- Create a `main.rb `file that will serve as your console app entry point.

- At the start, should:

- Present the user with a list of options to perform and let a user choose an option.
- Have a way to quit the app.

- The following options should be available:

```
List all books
List all music albums
List all movies
List of games
List all genres (e.g 'Comedy', 'Thriller')
List all labels (e.g. 'Gift', 'New')
List all authors (e.g. 'Stephen King')
List all sources (e.g. 'From a friend, 'Online shop')
Add a book
Add a music album
Add a movie
Add a game
```

- All data should be preserved by saving collections in .json files.

3. **Database**

- Create a schema.sql file with tables that will be analogical to the structure of the classes in your app. As you cannot implement inheritance in the database tables - add all properties and associations from the parent Item class as table columns to all tables based on the child classes. Thanks to that it can be used to store data in the future.

4. **Project documentation**

- [Video Presentation](https://drive.google.com/file/d/1oKnPJmn8yeQWsavgBmri5uMlPWN26qNH/view?usp=sharing)

### CODE REVIEW

**Review by FrederickMih — CHANGES_REQUESTED** [review]
## Hi :raising_hand: @NahnahAJ, @NatanimA, @arabiu033 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights :green_circle: 

- Data from JSON was loaded when the app started :+1: 
- All the requested tables in DB were created :+1: 
- Great README.md file :+1: 
- PR is professional :+1: 
- All Linter checks passed :+1: 


## Required Changes ♻️

- [ ] **Video Presentations:** Presentation of this project was fantastic and almost took 15 minutes. Based on the requirements of this project, the maximum length of the video should be 5 minutes. Let's kindly meet the project requirements.


_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: item.rb [hunk 1]
```diff
@@ -0,0 +1,41 @@
+require 'date'
+class Item
+  attr_accessor :source, :publish_date, :self, :label
```
Comment (addod19) [inline item.rb:3]: - [ ] Nice job addressing the comment from the previous code review, however, in addressing the comment we left out the `:label`. Let us kindly fix it

**Review by addod19 — CHANGES_REQUESTED** [review]
### Hi TEAM, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights
- [x] Great job testing all the methods :rocket: 

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by elisha2kyakpo1 — APPROVED** [review]
Hi 👋 team, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights

### 👍 You have worked so hard, it's time to celebrate 🎉🎉🎉

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**